src/core/debug_state.cpp

```diff
@@ -142,8 +143,7 @@ void DebugStateImpl::PushQueueDump(QueueDump dump) {
     frame.queues.push_back(std::move(dump));
 }
 
-void DebugStateImpl::PushRegsDump(uintptr_t base_addr, uintptr_t header_addr,
-                                  const AmdGpu::Liverpool::Regs& regs, bool is_compute) {
+void DebugStateImpl::PushRegsDump(uintptr_t base_addr, uintptr_t header_addr, bool is_compute) {
```

Comment:
To avoid reliance on global state this can be split into separate functions for gfx and compute (PushComputeRegsDump) which receive regs ref and ComputeProgram ref respectively. I'm not actually sure if compute dump relies on full regs